array/view.mbt

```diff
@@ -27,3 +27,262 @@ pub fn each[T](self : ArrayView[T], f : (T) -> Unit) -> Unit {
     f(self[i])
   }
 }
+
+///|
+pub fn rev_each[T](self : ArrayView[T], f : (T) -> Unit) -> Unit {
+  let len = self.length()
+  for i in 0..<len {
+    f(self[len - i - 1])
+  }
+}
+
+///|
+pub fn eachi[T](self : ArrayView[T], f : (Int, T) -> Unit) -> Unit {
+  for i, v in self {
+    f(i, v)
+  }
+}
+
+///|
+pub fn rev_eachi[T](self : ArrayView[T], f : (Int, T) -> Unit) -> Unit {
+  let len = self.length()
+  for i in 0..<len {
+    f(i, self[len - i - 1])
+  }
+}
+
+///|
+/// Compares two arrayviews for equality of items, ignoring start/end positions.
+pub fn op_equal[T : Eq](self : ArrayView[T], other : ArrayView[T]) -> Bool {
```

Comment:
@FlammeShadow Could you change this to `impl[T] Eq for ArrayView[T] ...`

Reply:
No problems.